contrib/platform/src/com/sun/jna/platform/win32/Kernel32Util.java

```diff
@@ -741,6 +744,8 @@ public static final String extractVolumeGUID(String volumeGUIDPath) {
             if (start == null) {
                 throw new IllegalStateException("LockResource returned null.");
             }
+            // have to capture it into a byte array before you free the library, otherwise bad things happen.
+            results = start.getByteArray(0, length);
```

Comment:
This was the mistake. If you don't get the resource before calling FreeLibrary() you're doomed.